http2/src/main/java/com/blazemeter/jmeter/http2/sampler/HTTP2StreamHandler.java

```diff
@@ -259,10 +237,14 @@ public void onReset(Stream stream, ResetFrame frame) {
     result.setResponseMessage(ErrorCode.from(frame.getError()).name());
     result.sampleEnd();
     result.setSuccessful(((frame.getError() == ErrorCode.NO_ERROR.code))
-        ||(frame.getError() == ErrorCode.CANCEL_STREAM_ERROR.code));
-    result.setPendingResponse(false);
+        || (frame.getError() == ErrorCode.CANCEL_STREAM_ERROR.code));
+    if (!result.isSecondaryRequest()) {
```

Comment:
since this logic and if !isSync is duplicated in several places maybe makes sense to move to a completeSample method in the result?